addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1178 @@
+        return self.write({'probability': 0, 'active': False})
+    # Backward compatibility
+    case_mark_lost = action_set_lost
+
+    @api.multi
+    def action_set_active(self):
+        return self.write({'active': True})
+
+    @api.multi
+    def action_set_unactive(self):
+        return self.write({'active': False})
+
+    @api.multi
+    def action_set_won(self):
+        """ Won semantic: probability = 100 (active untouched) """
+        stages_leads = {}
+        for lead in self:
+            stage_id = lead.stage_find(lead.team_id.id or False, [('probability', '=', 100.0), ('on_change', '=', True)])
+            if stage_id:
+                if stages_leads.get(stage_id):
+                    stages_leads[stage_id].append(lead.id)
+                else:
+                    stages_leads[stage_id] = [lead.id]
+        for stage_id, lead_ids in stages_leads.items():
+            self.browse(lead_ids).write({'stage_id': stage_id})
+        return self.write({'probability': 100})
+    # Backward compatibility
+    case_mark_won = action_set_won
+
+    @api.multi
+    def log_next_activity_1(self):
+        return self.set_next_activity(next_activity_name='activity_1_id')
+
+    @api.multi
+    def log_next_activity_2(self):
+        return self.set_next_activity(next_activity_name='activity_2_id')
+
+    @api.multi
+    def log_next_activity_3(self):
+        return self.set_next_activity(next_activity_name='activity_3_id')
+
+    def set_next_activity(self, next_activity_name):
+        for lead in self:
+            if not lead.last_activity_id:
+                continue
+            next_activity = next_activity_name and getattr(lead.last_activity_id, next_activity_name, False) or False
+            if next_activity:
+                date_action = False
+                if next_activity.days:
+                    date_action = fields.Datetime.to_string(fields.Datetime.now() + timedelta(days=next_activity.days))
+                lead.write({
+                    'next_activity_id': next_activity.id,
+                    'date_action': date_action,
+                    'title_action': next_activity.description,
+                })
+        return True
+
+    @api.multi
+    def log_next_activity_done(self, next_activity_name=False):
+        to_clear_ids = []
+        for lead in self:
+            if not lead.next_activity_id:
+                continue
+            body_html = """<div><b>${object.next_activity_id.name}</b></div>
+%if object.title_action:
+<div>${object.title_action}</div>
+%endif"""
+            body_html = self.env['mail.template'].render_template(body_html, 'crm.lead', lead.id)
+            lead.message_post(body_html, subtype_id=lead.next_activity_id.subtype_id.id)
+            to_clear_ids.append(lead.id)
+            lead.write({'last_activity_id': lead.next_activity_id.id})
+
+        if to_clear_ids:
+            self.cancel_next_activity(to_clear_ids)
```

Comment:
wrong. call cancel_next_activity with recordset(s). first browse then call. 
like  this way : 
self.browse(to_clear_ids).cancel_next_activity()